src/main/java/org/jabref/logic/search/SearchQuery.java

```diff
@@ -124,6 +127,18 @@ public boolean isRegularExpression() {
 
     // Returns a regular expression pattern in the form (w1)|(w2)| ... wi are escaped if no regular expression search is enabled
     public Optional<Pattern> getPatternForWords() {
+        return joinWordsToPattern(false);
+    }
+
+    // Returns a regular expression pattern in the form (w1)|(w2)| ... wi are escaped for javascript if no regular expression search is enabled
+    public Optional<Pattern> getJsPatternForWords() {
+        return joinWordsToPattern(true);
+    }
+
+    /* Returns a regular expression pattern in the form (w1)|(w2)| ... wi are escaped if no regular expression search is enabled
+     * @param escapeSpecialCharsForJS whether to escape characters in wi for javascript regexp (escaping all special characters) or for java (using \Q and \E)
+     */
+    private Optional<Pattern> joinWordsToPattern(boolean escapeSpecialCharsForJS) {
```

Comment:
Better is to use an enum here, eg.. EscapMode, with java and javascript as values:
https://www.teamten.com/lawrence/programming/prefer-enums-over-booleans.html